### PR TITLE
Maestro Scenario Tests - Add cloning uber test

### PR DESCRIPTION
**Pull request by meganaquinn**
- Add the C# version of the cloning ubertest

### CODE REVIEW

File: src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs [hunk 1]
```diff
@@ -736,6 +736,22 @@ public async Task<TemporaryDirectory> CloneAzDoRepositoryAsync(string repoName,
             return shareable.TryTake()!;
         }
 
+        public async Task<TemporaryDirectory> CloneRepositoryWithDarc(string repoName, string version, string reposToIgnore, bool includeToolset, int depth)
+        {
+            string sourceRepoUri = GetRepoUrl("dotnet", repoName);
+
+            using var shareable = Shareable.Create(TemporaryDirectory.Get());
+            string directory = shareable.Peek()!.Directory;
```
Comment (ChadNedzlek) [inline src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs:744]: I thought we had nullable off in arcade-services...  We shouldn't have/need this ! here.

File: src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                {
+                    string path = Path.Join(clonedReposFolder, name);
+                    DirectoryAssert.Exists(path, $"Expected cloned repo '{name}' but not found at {path}");
+
+                    string versionPath = Path.Join(path, "eng", "Version.Details.xml");
+                    FileAssert.Exists(versionPath, $"Expected a file at {versionPath}");
+
+                    using (FileStream stream = File.OpenRead(versionPath))
+                    {
+                        SHA256 hashGenerator = SHA256.Create();
```
Comment (ChadNedzlek) [inline src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs:145]: This is disposable too
```suggestion
                        using SHA256 hashGenerator = SHA256.Create();
```

File: src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                        byte[] fileHash = hashGenerator.ComputeHash(stream);
```
Comment (ChadNedzlek) [inline src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs:146]: It's... really odd that we hash this file.  It'll make any failure more or less impossible to diagnose, and I'd probably just replace the hash in the test with whatever it said... but that's what the old test did, I suppose. :-(

File: src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                        string fileHashInHex = BitConverter.ToString(fileHash).Replace("-", "");
+                        Assert.AreEqual(expectedRepos[name], fileHashInHex, $"Expected {versionPath} to have hash '{expectedRepos[name]}', actual hash '{fileHash}'");
```
Comment (ChadNedzlek) [inline src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs:148]: Are none of the tests using FluentAssertions? I didn't really notice this before.

File: src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs (hunk 3 above)
Comment (meganaquinn) [reply src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs:146]: That's basically what I had to do - I'm working on the assumption that the files I'm getting in development are the right ones & set up the hashes accordingly. I have an earlier version of the test that used the FileInfo, but I seem to have deleted the test side of it without pushing, so I have no idea where I was getting the FileInfo for comparison. I'm open to suggestions for a future iteration.

File: src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs (hunk 4 above)
Comment (meganaquinn) [reply src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Clone.cs:148]: I don't think any of the scenario tests are, no. Most of them were written before the whole test framework merge you did, so they're just NUnit.

File: src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs (hunk 1 above)
Comment (meganaquinn) [reply src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs:744]: I don't know, I've just been reusing the base code that Alex set up for the initial test. I'll give it a shot.

**Review by ChadNedzlek — APPROVED** [review]

**Pull request merged**